src/documentation/io/authoring-overview.md

```diff
@@ -10,35 +10,76 @@ permalink: /documentation/io/authoring-overview/
+2. If you are planning to contribute your I/O transform to the Beam community, you'll be going through the normal Beam contribution life cycle - see the [Apache Beam Contribution Guide]({{ site.baseurl }}/contribute/contribution-guide/) for more details.
+3. As you're working on your IO transform, the [PTransform Style Guide]({{ site.baseurl }}/contribute/ptransform-style-guide/) will provide specific information on many implementation details of transform. **Much of the information in the PTransform Style Guide is directly relevant to writing I/O Transforms**.
+
 
 ## Read transforms
-TODO
+Read transforms take data from outside of the Beam pipeline and produce PCollections of data.
+
+For data stores or files types where the data can be read in parallel, you can actually think of the process as a mini-pipeline. This will often consist of two steps:
+1. Splitting the data into parts to be read in parallel
+2. Reading from each of those parts.
+
+Each of those steps will be a ParDo, with a GroupByKey in between. The GroupByKey is an implementation detail but for most runners it allows the runner to use different numbers of workers for determing how to split up the data to be read into chunks (which will likely occur on very few workers) and reading (which will likely benefit from more workers.) For runners which support Dynamic Work Rebalancing, the GBK will also allow that to occur.
+
+Here are some examples of read transform implementations using the "reading as a mini-pipeline" model to use when data can be read in parallel:
+* **Reading from chunks of a file** - if you have a file format that can be sub-divided, the structure will look like:
+  * Determine Byte Ranges ParDo: As input, takes in a file name. Emit a PCollection of byte ranges inside of the file.
+  * Read Byte Range ParDo: Given the PCollection of byte ranges, and reads each byte range and reads the data in that byte range, emitting a PCollection of records.
+* **Reading from a file glob** - For example reading all files in "~/data/**"
+  * Get File Paths ParDo: As input, take in a file glob. Emit a PCollection of strings, each of which is a file path.
+  * Reading ParDo: Given the PCollection of file paths, read each one, emitting a PCollection of records.
+* **Reading from a NoSQL Database** (eg Apache HBase) - these databases often allow reading from ranges in parallel.
+  * Determine Key Ranges ParDo: As input, receive connection information for the database and the key range to read from. Output a PCollection of key ranges that can be read in parallel efficiently.
+  * Read Key Range ParDo: Given the PCollection of key ranges, read the key range, emitting a PCollection of records.
+
+For data stores or files where reading cannot occur in parallel, reading is a simple task that can be accomplished with a single ParDo. For example:
```

Comment:
You should also stick a GBK/reshuffle on top of it, rather than relying on the user doing it after applying your transform.